From 5003bd25c9a75dee4730dcef75305e9a1b7c0c63 Mon Sep 17 00:00:00 2001
From: Javier Celaya <javier.celaya@flexvdi.com>
Date: Fri, 26 Jan 2018 10:08:03 +0100
Subject: [PATCH 4/4] Fix setting appsrc properties in i686 arch

The "max-bytes" property of the appsrc GStreamer element expects a
64-bit value. Calling:
g_object_set(source, "max-bytes", 0, NULL);
in a 32-bit architecture ends up with a wrong value, it must be cast to
a long long int:
g_object_set(source, "max-bytes", 0LL, NULL);
---
 src/channel-display-gst.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/channel-display-gst.c b/src/channel-display-gst.c
index d7f47d1..ff4a884 100644
--- a/src/channel-display-gst.c
+++ b/src/channel-display-gst.c
@@ -333,7 +333,7 @@ static void app_source_setup(GstElement *pipeline G_GNUC_UNUSED,
                  "caps", caps,
                  "is-live", TRUE,
                  "format", GST_FORMAT_TIME,
-                 "max-bytes", 0,
+                 "max-bytes", G_GINT64_CONSTANT(0),
                  "block", TRUE,
                  NULL);
     gst_caps_unref(caps);
-- 
2.14.3

